### ISSUE TITLE
Security vulnerability detected in nokogiri gem

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#2006]
From Github:

> We found a potential security vulnerability in one of the dependencies used by a repository that you contribute to.
> 
> Known critical severity security vulnerability detected in nokogiri < 1.8.1 defined in Gemfile.lock. 
> Gemfile.lock update suggested: nokogiri ~> 1.8.1. 
> Always verify the validity and compatibility of suggestions with your codebase. 	

## Severity

Unknown. Depends on how we use nokogiri.

## Possible Fix

`bundle update nokogiri`

**Comment by mkllnk** [comment]
We can't update nokogiri on it's own:
```
Bundler could not find compatible versions for gem "nokogiri":
  In Gemfile:
    nokogiri (>= 1.8.1)

    deface was resolved to 1.0.0.rc3, which depends on
      nokogiri (~> 1.6.0)

    deface was resolved to 1.0.0.rc3, which depends on
      nokogiri (~> 1.6.0)
```

**Comment by sauloperez** [comment]
> So we need to update spree and db2fog so that we can update rails so that we can update deface so that we can update nokogiri. Welcome to ruby gems.

Welcome to software libraries :trollface:

**Comment by daniellemoorhead** [comment]
#operationfixthebugs

Right then, as the only `bugs-s1` issue currently open this here bug needs to get busted, and as soon as humanly possible.

@myriamboure @kirstenalarsen @oeoeaio @enricostano as the product curation team do y'all agree that this is the next most important thing to put through the delivery pipe and if so can we please assign people to start working on it?

Ping @mkllnk @sauloperez also, given you guys have history with this one.

**Comment by sauloperez** [comment]
I wasn't aware of that S1 but this can't be done in reasonable time span. It's a yak shaving of enourmous dimension :trollface:. I'm not suggesting a lower severity; I just want us all to be aware.

**Comment by mkllnk** [comment]
Blocked by Spree upgrade, I would say. Which would make Spree upgrade S1, correct?

**Comment by sauloperez** [comment]
Wasn't already an S1 :trollface: .

**Comment by mkllnk** [comment]
Without further study of our nokogiri use and how the vulnerabilities can be exploited it's difficult to tell. If this vulnerability means that anyone in the world can hack our servers over the network, that's a very critical bug. If it's actually not exploitable, but slows down our development, then it's only tech debt.

Since we can't do anything about it at the moment, it doesn't really matter. We need to get the Spree update done and update our dependencies.

**Comment by sauloperez** [comment]
Yes, that's the high price of not keeping dependencies up to date

**Comment by luisramos0** [comment]
Removing the reference to #2109 and moving it to epic #2584 (all the issues blocked by the spree upgrade).

**Comment by luisramos0** [comment]
This fix is still going to take a little while to come...

Having a security issue well indexed by google is not a very wise approach. I think we should delete this issue and manage this in some other way. what do you think?

**Comment by mkllnk** [comment]
I think if Github has an automated way of finding this vulnerability then others have that as well. We are open source and there is no secret about our dependencies. We can't hide it.

About the impact: I'm not aware of any user input that is passed to nokogiri. Without that connection OFN is not vulnerable. We should still fix it as soon as possible though.

**Comment by luisramos0** [comment]
I am not sure I follow @mkllnk 
If ofn is not vulnerable why is this an S1?
our dependencies are open, but making our security vulnerabilities like this one open and documented for the world is optional. We can delete this issue and that reduces visibility over our security vulnerabilities.

**Comment by mkllnk** [comment]
> If ofn is not vulnerable why is this an S1?

We don't know for sure. I haven't spend much time looking into it because I think we should invest that time into upgrading. The potential impact is very high though. It's better to assume the worst case.

Yes, we can delete the issue and it would reduce the visibility. But I guess that the more likely scenario would be that somebody who knows the vulnerability very well scans Github for all projects using this dependency and then tries to exploit them. I think the risk of someone stumbling across this issue and then being motivated to research and exploit a vulnerability that may not even be exploitable is low. But the visibility of this issue helps us prioritise upgrades which will actually solve the issue.

That's just my opinion. Lots of guesses in there and a difficult trade-off. If more people think we should delete the issue, I'm okay with that.

**Comment by Matt-Yorkley** [comment]
I vote for deleting this issue, any users with write permission already get a vulnerability reminder in the main repo page.

**Comment by luisramos0** [comment]
thanks Maikel for explaining! I understand what you saying and I agree it's not a major thing to have this issue here or not. Still I vote to delete the issue and start managing all security related topics in a private manner (as we already do for some other things).

**Comment by mkllnk** [comment]
Okay. Unfortunately, the nokogiri alerts have been dismissed and can't be activated again. We just need to keep it in our minds. I will try to delete this issue.
